app/models/bet.rb

```diff
@@ -2,4 +2,41 @@ class Bet < ApplicationRecord
   has_many :users, through: :user_bets
   has_many :user_bets
   has_many :options, through: :user_bets, source: :bet_option
+
+  def resolve
+    losers.each do |debtor|
+      debt = amount_bet(debtor)
+      win_contributions.each do |contribution|
+        debtor.debts.create(creditor: contribution[:creditor],
+                            amount:   debt * contribution[:percentage])
+      end
+    end
+  end
+
+  def win_option
+    options.winners
+  end
+
+  def winner_total
+    user_bets.winners.map(&:amount_bet).sum
+  end
+
+  def winners
+    users.winners
+  end
+
+  def losers
+    users.losers
+  end
+
+  def amount_bet(user)
```

Comment:
I think this is a really good method. It is semantic (the name indicates what it does) it takes a reasonable argument and it does 1 thing and returns the output of that one thing. 

I am not sure however that this is functionality that is needed. I think a different approach would be to iterate over the user bets without caring specifically which user they belong to